Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -56,10 +58,42 @@ def check_python_framework_links(lib)
     EOS
   end
 
+  def check_linkage
+    return unless formula.prefix.directory?
+    keg = Keg.new(formula.prefix)
+    checker = LinkageChecker.new(keg, formula)
+    s = ""
+
+    if checker.broken_dylibs?
+      s += <<-EOS.undent
+        Broken dylib links found:
+          #{checker.broken_dylibs.to_a * "\n          "}
+
+        It is likely the installation was broken.
+        Please check formula and make sure it is built from source.
+      EOS
+    end
+
+    if checker.undeclared_deps?
+      s += <<-EOS.undent
+        Undeclared dependencies found:
+          #{checker.undeclared_deps * "\n          "}
+
+        Formulae are required to declare all linked dependencies.
+        Please check whether these dependencies are indeed needed.
+        If so, please add all missing dependencies to the formula with:
```

Comment:
If contributors are going to pre-add these in their PRs (rather than waiting for CI to tell them what to add), the procedure for that should be for the contributor to run brew test-bot in a totally empty prefix, unless we want to be somehow (magically) policing this ourselves (which means us doing the test-bot runs locally since the CI results won't be available since the contributor already over-added, which means brew audit prints nothing and the extra junk makes it to master).